File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        map<int,int> m;
        for(int i=0;i<arr.size();i++){
            m[arr[i]] = 1;
        }
        int r = 1;
        for(auto x:m){
            m[x.first] = r;
            r++;
        }
        for(int i=0;i<arr.size();i++){
            arr[i] = m[arr[i]];
        }
        return arr;
    }
};

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        TreeMap<Integer,Integer> m = new TreeMap<>();
        for(int i=0;i<arr.length;i++){
            m.put(arr[i],-1);
        }
        int r = 1;
        for(Map.Entry<Integer,Integer> mp : m.entrySet()){
            m.put(mp.getKey(),r);
            r++;
        }
        
        for(int i=0;i<arr.length;i++){
            arr[i] = m.get(arr[i]);
        }
        
        return arr;
    }
}